### ISSUE TITLE
Sketch-Guided Text-to-Image Diffusion Models

### ISSUE THREAD

**Issue by nmhkahn** [nmhkahn/paper-reading#46]
## Motivation
- 텍스트 프롬프트만 사용하는 방식은 생성 이미지의 공간적인 정보를 충분히 반영하지 못함
- 작은 MLP를 학습시켜서 spatial map을 latent guide에 사용하도록 모델 설계

## Method
- 제안하는 방법은 1) latent edge predictor 학습 2) edge predictor를 SD 모델 인퍼런스 시 guidance로 사용 의 2가지 스텝으로 구성
- Latent edge predictor는 pre-trained SD에 스케치 guidance를 주기 위한 네트워크로, per-pixel MLP network 형태
    - UNet의 internal activation map을 입력받아서 edge map을 생성하도록 학습 (아래 그림)
    - <img width="350" alt="Screenshot 2022-12-22 at 7 26 11 PM" src="https://user-images.githubusercontent.com/16869368/209114065-348d9253-cb1e-4ace-98e7-96406f61c55a.png">
    - UNet의 $l$개 레이어에서 추출한 activation map + noise level (step) $t$에 해당하는 positional encoding을 입력으로 받음
    - 이 edge predictor만 별도로 학습. Ground-truth edge map은 기존 edge extractor 사용해서 이미지로부터 추출
- 학습된 edge predictor는 아래 그림처럼 인퍼런스 시에 추가적인 guidance에 사용
    - <img width="350" alt="Screenshot 2022-12-22 at 7 29 54 PM" src="https://user-images.githubusercontent.com/16869368/209114768-4d7c2a55-c1ae-4e17-a485-8bcbcc83f0af.png">
    - Activation map으로 부터 생성된 edge가 유저가 제공한 target edge를 따라하도록 loss 계산 -> sampling guidance
    - GLIDE에서 사용한 anti-gradient 사용하고, 초반 50%의 step에만 guidance <- 아마 후반 sampling step 에서는 object shape 보다 디테일 생성 위주이기 때문일듯

**Comment by nmhkahn** [comment]
## Discussion
- ILVR (#45) 이나 이 논문처럼 sampling 시에 guidance를 주는 방법을 SD에 결합하는 것도 좋은 연구 방향이라고 생각
